sdk/eventhub/event-hubs/test/internal/auth.spec.ts

```diff
@@ -2,195 +2,164 @@
 // Licensed under the MIT license.
 
 import { AzureNamedKeyCredential, AzureSASCredential } from "@azure/core-auth";
-import { EnvVarKeys, getEnvVars } from "../public/utils/testUtils";
 import {
-  EventHubConnectionStringProperties,
-  EventHubConsumerClient,
-  EventHubProducerClient,
   parseEventHubConnectionString,
-} from "../../src";
-import chai from "chai";
-import { createMockServer } from "../public/utils/mockService";
-import { createSasTokenProvider } from "@azure/core-amqp";
-import { testWithServiceTypes } from "../public/utils/testWithServiceTypes";
-
-const should = chai.should();
-
-testWithServiceTypes((serviceVersion) => {
-  const env = getEnvVars();
-  if (serviceVersion === "mock") {
-    let service: ReturnType<typeof createMockServer>;
-    before("Starting mock service", () => {
-      service = createMockServer();
-      return service.start();
-    });
-
-    after("Stopping mock service", () => {
-      return service?.stop();
-    });
+  EventHubConsumerClient,
+  type EventHubProducerClient,
+} from "../../src/index.js";
+import {
+  createConsumer,
+  createProducer,
+  getConnectionStringWithKey,
+  getConnectionStringWithSAS,
+} from "../utils/clients.js";
+import { isSasTokenProvider } from "@azure/core-amqp";
+import { should, assert } from "../utils/chai.js";
+import { describe, it, afterEach, beforeEach, beforeAll } from "vitest";
+
+function getCredential(client: EventHubConsumerClient | EventHubProducerClient): any {
+  const cred =
+    // typescript doesn't allow access to a shared member between the two classes.
+    client instanceof EventHubConsumerClient
+      ? client["_context"].tokenCredential
+      : client["_context"].tokenCredential;
+  if (!isSasTokenProvider(cred)) {
+    assert.fail("Expected a SasTokenProviderImpl instance.");
   }
+  // cred is of type SasTokenProviderImpl which is not exported.
+  return (cred as any)["_credential"];
+}
+
+describe.skipIf(!getConnectionStringWithKey())("Authentication via", function () {
```

Comment:
If `EVENTHUB_CONNECTION_STRING` environment variable is not available in live mode.

Reply:
I believe we could still live-test connection string for functionalities that relies on it.

Reply:
Yes, all you need to do is to provide one in the environment variable.

Reply:
I think we should still throw if one is not provided in live mode, otherwise we won't notice the skipped test due to mis configured environtment.  If there are other places that already have that then it's fine to skip here.

Reply:
I discussed this with Jeff and he pointed out the same pattern is being used in storage: https://github.com/Azure/azure-sdk-for-js/blob/70ea132759e015e60d45582c815e706976ae50fd/sdk/storage/storage-blob/test/blobserviceclient.spec.ts#L471-L477

Overall, I think it is a good pattern and it is especially relevant today when you may want to test but don't want to mess with secrets. The test also provides good coverage even in the absence of connection strings.

Reply:
If we expect this test to run under live mode I'd prefer it to throw instead of being skipped. But no strong feeling.